### PR TITLE
Tickets

### PR DESCRIPTION
**Pull request by sinkingsugar**
Big refactor for tickets

### CODE REVIEW

File: pallets/protos/src/lib.rs [hunk 1]
```diff
@@ -809,25 +831,46 @@ pub mod pallet {
...
+					match license {
+						UsageLicense::NotAvailable => {
+							return Err(Error::<T>::Unauthorized.into());
+						},
+						UsageLicense::Available => { continue },
+						UsageLicense::Tickets(amount) => {
+							let balance = <pallet_assets::Pallet<T> as Inspect<T::AccountId>>::balance(T::TicketsAssetId::get(), &who);
+							let amount: u64 = amount.into();
+							ensure!(balance >= amount.saturated_into(), Error::<T>::InsufficientBalance);
+							// TODO transfer tickets
```
Comment (alebaffa) [inline pallets/protos/src/lib.rs]: What do you mean with _transfer tickets_?

Comment (sinkingsugar) [reply pallets/protos/src/lib.rs]: wip, was wrong

**Comment by sinkingsugar** [comment]
I think it's done, someone will need to expand it with lot of tests, you guys decide who will do it as follow up please! 🙏

**Comment by sinkingsugar** [comment]
Lot's of formatting.. because we merged PRs without running `cargo fmt` please make it an habit...

**Comment by sinkingsugar** [comment]
@alebaffa can you also takeover the substrate fork with untradeable assets? Should be done too ish, some weird check/test was not working.
Also from next month we might wanna use `polkadot-XXX` tags. Let's discuss it early in the week please 🙏

File: runtime/src/lib.rs [hunk 2]
```diff
@@ -379,12 +379,16 @@ impl pallet_accounts::Config for Runtime {
...
+	pub const TicketsAssetId: u64 = 1337;
+}
+
 impl pallet_protos::Config for Runtime {
 	type Event = Event;
 	type WeightInfo = ();
 	type StorageBytesMultiplier = StorageBytesMultiplier;
-	// type StakeLockupPeriod = ConstU64<100800>; // one week
-	type StakeLockupPeriod = ConstU64<5>; // one week
+	type CurationExpiration = ConstU64<100800>; // one week
```
Comment (aang114) [inline runtime/src/lib.rs:390]: Giovanni, why do all curations expire after a certain amount of time?

File: pallets/protos/src/lib.rs [hunk 3]
```diff
@@ -137,24 +151,33 @@ pub mod pallet {
...
+		+ pallet_contracts::Config
 	{
 		/// Because this pallet emits events, it depends on the runtime's definition of an event.
 		type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
 
 		type WeightInfo: WeightInfo;
 
+		/// Const per byte multiplier
 		#[pallet::constant]
 		type StorageBytesMultiplier: Get<u64>;
```
Comment (aang114) [inline pallets/protos/src/lib.rs:173]: I have no idea what this does by the way 😅. But it's okay. I'll figure it out later

**Review by aang114 — APPROVED** [review]
Everything looks correct! 

I don't like the word "curate" though 😅 (cos it doesn't correctly describe what exactly is happening 😅). But we can change the word later, if needed

File: runtime/src/lib.rs (hunk 2 above)
Comment (aang114) [reply runtime/src/lib.rs:390]: @sinkingsugar

**Comment by aang114** [comment]
I just added a few lines of documentation to this branch and now this branch fails !??!? Did anyone (@sinkingsugar @alebaffa ) change the CI code recently - because this doesn't make sense 😅

**Comment by aang114** [comment]
> I just added a few lines of documentation to this branch and now this branch fails !??!? Did anyone (@sinkingsugar @alebaffa ) change the CI code recently - because this doesn't make sense 😅

The "push" CI seems to be working. So this is truly weird. **@sinkingsugar @alebaffa you might want to take note of this 😅**

File: pallets/protos/src/lib.rs (hunk 3 above)
Comment (sinkingsugar) [reply pallets/protos/src/lib.rs:173]: Increases the weight of data bytes, limits storage amount x block/extrinsic

File: runtime/src/lib.rs (hunk 2 above)
Comment (sinkingsugar) [reply runtime/src/lib.rs:390]: Well we don't want to have people curate once and their curation be valid forever, it's a renewal game.

**Review by alebaffa — APPROVED** [review]

File: pallets/protos/src/lib.rs (hunk 3 above)
Comment (aang114) [reply pallets/protos/src/lib.rs:173]: `StorageBytesMultiplier` is basically the weight for adding a byte worth of `data` (the last parameter), yes?

Comment (aang114) [reply pallets/protos/src/lib.rs:173]: I'm documenting it like this in another PR:

/// Weight for adding a a byte worth of storage in certain extrinsics such as `upload()`.

**Pull request merged**